cmd/touch/touch.go

```diff
@@ -0,0 +1,69 @@
+package touch
+
+import (
+	"bytes"
+	"time"
+
+	"github.com/ncw/rclone/cmd"
+	"github.com/ncw/rclone/fs"
+	"github.com/spf13/cobra"
+)
+
+var (
+	notCreateNewFile bool
+	timeAsArgument   string
+)
+
+func init() {
+	cmd.Root.AddCommand(commandDefintion)
+	flags := commandDefintion.Flags()
+	flags.BoolVarP(&notCreateNewFile, "not-create", "C", false, "Do not create the file if it does not exist.")
+	flags.StringVarP(&timeAsArgument, "time", "t", "", "Change the modification times to the specified time instead of the current time of day. The argument is of the form 'YYMMDD' (ex. 17.10.30) or 'YYYY-MM-DDTHH:MM:SS' (ex. 2006-01-02T15:04:05)")
+}
+
+var commandDefintion = &cobra.Command{
+	Use:   "touch remote:path",
+	Short: `Create new file or change file modification time.`,
+	Run: func(command *cobra.Command, args []string) {
+		cmd.CheckArgs(1, 1, command, args)
+		fsrc, srcFileName := cmd.NewFsDstFile(args)
+		cmd.Run(true, false, command, func() error {
+			return Touch(fsrc, srcFileName)
+		})
+	},
+}
+
+//Touch create new file or change file modification time.
+func Touch(fsrc fs.Fs, srcFileName string) error {
+	timeAtr := time.Now()
+	if timeAsArgument != "" {
+		layout := "060102"
+		if len(timeAsArgument) == len("2006-01-02T15:04:05") {
+			layout = "2006-01-02T15:04:05"
+		}
+		timeAtrFromFlags, err := time.Parse(layout, timeAsArgument)
+		if err == nil {
+			timeAtr = timeAtrFromFlags
+		}
+	}
+	file, err := fsrc.NewObject(srcFileName)
+	if err != nil {
+		_, err = fsrc.List(srcFileName)
```

Comment:
What is this List for?

Reply:
I did not find how to check the object type. When this function return error, that implies object in srcFileName is not a directory. I know, it isn't a correct way to check it. Without check it rclone create file, even if directory with this name exists. In this case linux' own command do nothing.  
Could you tell me how to check if file is not a directory :)?

Reply:
If you call NewObject you will always get an Object not a Directory.

Reply:
Linux command 'touch' also updates directory's timestamp. Should I implement this functionality? If not, function can always create new file and I don't need to check if folder exists.  In case directory with this name exists, nothing will happen, but google drive apart other clouds allow users to have folder and file with this same name (in this same directory) and because of it function will create new objects (in google drive).
If rclone does not allow to change directory modifications time it can stay as it is now, well:
If file exists - set modification time to now (if time is not specified)
else
try create empty file (now function doesn't return error if can't create new file (as linux command).

Reply:
rclone doesn't currently allow setting of modtimes on directories.  There is an issue about this (of course!) #1669 - I made a note on that issue to update rclone touch at that point too.